torch/functional.py

```diff
@@ -324,6 +325,118 @@ def parse_subscript(n: int) -> str:
+
+        if dtype_is_int:
+            pad_spec_is_int = (not pad_spec_tensor.is_floating_point()) and (not pad_spec_tensor.is_complex())
+            assert pad_spec_is_int, f"Expected '{arg_name}' to be of signed integer type"
+
+        pad_spec_tensor = pad_spec_tensor.to(dtype)
+
+    return pad_spec_tensor
+
+
+def pad(input, pad_width, mode='constant', *, constant_values=None, out=None):
+    r"""Pad a tensor by extending the size of one or more dimensions. The new
+    elements are filled with values according to the chosen padding mode.
+    Padding can be added to the beginning of each dimension, called "before
+    padding", and/or the end of each dimension, called "after padding".
+
+    Args:
+        input (Tensor): Tensor to pad
+
+        pad_width (Tensor, sequence, array_like, or int): Number of values
+            padded to the edges of each dimension. If a tensor is given,
+            it must be on CPU. The following formats can be used:
+
+                * Scalar, (all,), or ((all,)) yields equal before and after pad
+                  widths for all dimensions.
+                * ((both_1), ..., (both_N)) yields unique padding for each
+                  dimension, but before and after pad widths within each
+                  dimension are equal.
+                * ((before, after),) or (before, after) yields unique before
+                  and after pad widths, but same for all dimensions.
+                * ((before_1, after_1), ..., (before_N, after_N)) yields unique
+                  before and after pad widths for each dimension.
+
+        mode (str, optional): Padding mode to use to fill padding elements. One
+            of the following string values can be used. Default: `constant`
+
+                * 'constant' (default): Pads with constant values, according to
+                  the :attr:`constant_values` argument.
+                * 'empty': Pads with undefined values from uninitialized memory.
+
+    Keyword args:
+        constant_values (Tensor, sequence, array_like, or scalar): Used in
+            ``mode='constant'`` to specify the padding values for each dimension.
+            The following formats can be used. Default: 0
+
+                * Scalar, (all,), or ((all,)) yields equal before and after pad
+                  values for all dimensions.
+                * ((both_1), ..., (both_N)) yields unique padding for each
+                  dimension, but before and after pad values within each
+                  dimension are equal.
+                * ((before, after),) or (before, after) yields unique before
+                  and after pad values, but same for all dimensions.
+                * ((before_1, after_1), ..., (before_N, after_N)) yields unique
+                  before and after pad values for each dimension.
+
+        out (Tensor, optional): Output tensor. Default: None
+
+    """
+    # TODO: I guess this should actually also check pad_width and constant_values,
```

Comment:
Would you elaborate on this comment? (in this github thread)

Reply:
I'm using `has_torch_function_unary` to check only `input` for torch_function. But `pad_width` and `constant_values` could also be tensors with a torch_function defined. It's probably simple to just mimic what other functions do in this case, but I just haven't gotten to it yet.

Reply:
I changed this to use `has_torch_function_variadic`, so that `input`, `pad_width`, and `constant_values` are checked.